### PR TITLE
Enable benchmarking all pallets for a given runtime with bench-all

### PR DESCRIPTION
**Pull request by seadanda**
Add a runtime parameter to bench-all to allow benchmarking all pallets for a given runtime with one command

### CODE REVIEW

**Comment by seadanda** [comment]
@mordamax I created this but for some reason the artifacts that are generated when it runs does not include a diff, only json output. Am I missing something?

**Comment by mordamax** [comment]
@seadanda could you please share the pipeline for the context? thx

**Comment by seadanda** [comment]
> @seadanda could you please share the pipeline for the context? thx

Ah I cleaned it and just used the individual bench scripts, I'm working on a branch for coretime, so I'll run it on that and update here when I have the artifacts to show you.

**Comment by seadanda** [comment]
@mordamax I realised that I sent the job link to someone in element, so I've got the job I ran for https://github.com/paritytech/polkadot-sdk/pull/2024 here: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4238416

**Comment by seadanda** [comment]
I've created https://github.com/paritytech/polkadot-sdk/pull/2548 and run it for `glutton-westend` to reproduce it on a small runtime.

**Comment by seadanda** [comment]
@mordamax The bench-bot run for glutton finished and didn't show any diff, so I ran `bench-all cumulus` with no runtime argument to compare, but actually it doesn't return a diff either for glutton. 

It did produce a diff for `asset-hub-rococo`, though so I ran bench bot with the runtime argument and it didn't produce a diff.

If you get a chance could you take a look please?

**Comment by mordamax** [comment]
@ggwpez i have no idea why that happens, may be you could help?
https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4546953
image

**Comment by seadanda** [comment]
Bump - sorry this is very old but has anybody got any ideas? It would be nice to merge this, but more importantly it seems that something is wrong with `bench-all` on `main` too.

**Comment by mordamax** [comment]
@seadanda i've updated your branch on sdk and re-started the bench with runtime specified to see the fresh results

File: commands/bench/lib/bench-all-cumulus.sh [hunk 1]
```diff
@@ -74,20 +76,57 @@ run_cumulus_bench() {
...
+    ;;
+    *)
+      echo "Unknown runtime: $runtime"
+      exit 1
+    ;;
+  esac
+
+  run_cumulus_bench $category $runtime $paraId
+
+  exit 0
```
Comment (mordamax) [inline commands/bench/lib/bench-all-cumulus.sh:113]: @seadanda this is the reason it doesn't go further and commit weights. just remove this exit 0

Comment (mordamax) [inline commands/bench/lib/bench-all-cumulus.sh:113]:
```suggestion
```

Comment (seadanda) [reply commands/bench/lib/bench-all-cumulus.sh]: ahh is this sourced into something else rather than run as a standalone script?

Comment (mordamax) [reply commands/bench/lib/bench-all-cumulus.sh]: yea, it's part of bench.sh, so supposed that it goes further to collect a diff
https://github.com/paritytech/command-bot-scripts/blob/main/commands/bench/bench.sh#L99
as this is common for all kinds of benches

Comment (mordamax) [reply commands/bench/lib/bench-all-cumulus.sh]: btw i removed and re-started bench to test it

**Comment by seadanda** [comment]
Looks like it worked: https://github.com/paritytech/polkadot-sdk/pull/2548/commits/bd477064e6587f61181e2049753af5b17645f101

Thanks for that - could we merge this in that case?

**Review by mordamax — APPROVED** [review]

**Comment by mordamax** [comment]
thanks for contribution @seadanda !

**Pull request merged**